src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Commitment.cs

```diff
@@ -43,5 +46,96 @@ public Commitment()
         public virtual ICollection<Apprenticeship> Apprenticeship { get; set; }
         public virtual ICollection<Message> Message { get; set; }
         public virtual ICollection<TransferRequest> TransferRequest { get; set; }
+
+        public virtual void AddDraftApprenticeship(DraftApprenticeshipDetails draftApprenticeshipDetails, IUlnValidator ulnValidator)
+        {
+            ValidateDraftApprenticeshipDetails(draftApprenticeshipDetails, ulnValidator);
+            var draftApprenticeship = new DraftApprenticeship(draftApprenticeshipDetails, Originator);
+            Apprenticeship.Add(draftApprenticeship);
+        }
+
+        private void ValidateDraftApprenticeshipDetails(DraftApprenticeshipDetails draftApprenticeshipDetails, IUlnValidator ulnValidator)
+        {
+            var errors = new List<DomainError>();
+            errors.AddRange(BuildNameValidationFailures(draftApprenticeshipDetails));
+            errors.AddRange(BuildEndDateValidationFailures(draftApprenticeshipDetails));
+            errors.AddRange(BuildCostValidationFailures(draftApprenticeshipDetails));
+            errors.AddRange(BuildReferenceValidationFailures(draftApprenticeshipDetails));
+                
+            errors.ThrowIfAny();
+        }
+
+        private IEnumerable<DomainError> BuildNameValidationFailures(DraftApprenticeshipDetails draftApprenticeshipDetails)
+        {
+
+            if (string.IsNullOrWhiteSpace(draftApprenticeshipDetails.FirstName))
```

Comment:
Do the multiple nested if...elses make this code hard to read?

Reply:
Yes, in fact they hid a small issue. Removed.